semver/validity.ts

```diff
@@ -83,76 +46,39 @@ export function isSemVerComparator(value: unknown): value is SemVerComparator {
  * @param value The value to check if its a valid SemVerRange
  * @returns True if its a valid SemVerRange otherwise false.
  */
-export function isSemVerRange(value: unknown): value is SemVerRange {
-  if (value == null) return false;
-  if (Array.isArray(value)) return false;
-  if (typeof value !== "object") return false;
-  const { ranges } = value as SemVerRange;
-  return (
-    Array.isArray(ranges),
-      ranges.every((r) =>
-        Array.isArray(r) && r.every((c) => isSemVerComparator(c))
-      )
-  );
-}
+export { isSemVerRange } from "./is_semver_range.ts";
 
 /**
+ * @deprecated (will be removed after 0.189.0) Import from `std/semver/is_valid_operator.ts` instead.
+ *
  * Checks to see if the value is a valid Operator string.
  *
  * Adds a type assertion if true.
  * @param value The value to check
  * @returns True if the value is a valid Operator string otherwise false.
  */
-export function isValidOperator(value: unknown): value is Operator {
-  if (typeof value !== "string") return false;
-  switch (value) {
-    case "":
-    case "=":
-    case "==":
-    case "===":
-    case "!==":
-    case "!=":
-    case ">":
-    case ">=":
-    case "<":
-    case "<=":
-      return true;
-    default:
-      return false;
-  }
-}
+export { isValidOperator } from "./is_valid_operator.ts";
 
 /**
+ * @deprecated (will be removed after 0.189.0) Import from `std/semver/is_valid_number.ts` instead.
+ *
  * Returns true if the value is a valid SemVer number.
  *
  * Must be a number. Must not be NaN. Can be positive or negative infinity.
  * Can be between 0 and MAX_SAFE_INTEGER.
  * @param value The value to check
  * @returns True if its a valid semver number
  */
-export function isValidNumber(value: unknown): value is number {
-  return (
-    typeof value === "number" &&
-    !Number.isNaN(value) && (
-      !Number.isFinite(value) ||
-      (0 <= value && value <= Number.MAX_SAFE_INTEGER)
-    )
-  );
-}
+export { isValidNumber } from "./is_valid_number.ts";
 
 /**
- * Returns true if the value is a valid semver prerelase or build identifier.
+ * @deprecated (will be removed after 0.189.0) Import from `std/semver/is_valid_string.ts` instead.
+ *
+ * Returns true if the value is a valid semver pre-release or build identifier.
  *
  * Must be a string. Must be between 1 and 256 characters long. Must match
  * the regular expression /[0-9A-Za-z-]+/.
  * @param value The value to check
  * @returns True if the value is a valid semver string.
  */
-export function isValidString(value: unknown): value is string {
-  return (
-    typeof value === "string" &&
-    value.length > 0 &&
-    value.length <= MAX_LENGTH &&
-    !!value.match(/[0-9A-Za-z-]+/)
-  );
-}
+export { isValidString } from "./is_valid_string.ts";
```

Comment:
I think we don't need this kind of re-export from subpath because currently the all APIs are only exported from `mod.ts` under `semver/`.

Reply:
Devs might be importing from files more directly rather than `mod.ts`. Removing these re-exports would break their setups without notice.

Reply:
I mean the file `validity.ts` doesn't exist in the current main. This is introduced in the rewrite

Reply:
If a function is exported and not prefixed with `_`, it probably should be added to `mod.ts`, as it's not private.